### PR TITLE
added error handling for eoas errors

### PR DESCRIPTION
**Pull request by mphprogrammer**
Added error handling for EOAS returned errors. Most errors from EOAS are returned with a 500 status and no minor codes. To figure out the exact nature of the error require parsing of the `<faultstring>` looking for tell-tale phrases to get more info. Also, some EOAS errors are returned with a status of 200, but a `<returnCode>500</returnCode>` in the body. This will handle both types, and return VA900.

Also stripping out tabs and newlines from body response, they affect the regexes that search for the errors. No valid EOAS response has tabs or newlines.

### CODE REVIEW

File: lib/preneeds/middleware/response/clean_response.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class CleanResponse < Faraday::Response::Middleware
         def on_complete(env)
           return unless env.url.to_s == Preneeds::Configuration.url
 
-          relevant_xml = env.body&.scan(%r{<S:Envelope[^<>]*>.*</S:Envelope[^<>]*>}i)&.first
+          relevant_xml = env.body&.gsub(/[\t\n]/, ' ')&.scan(%r{<S:Envelope[^<>]*>.*</S:Envelope[^<>]*>}i)&.first
```
Comment (saneshark) [inline lib/preneeds/middleware/response/clean_response.rb]: as we previously noted, please remove any '%' from the body as well.

File: lib/preneeds/middleware/response/eoas_xml_errors.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+module Preneeds
+  module Middleware
+    module Response
+      class EoasXmlErrors < Faraday::Response::Middleware
+        include SentryLogging
+        attr_reader :status, :fault, :code, :detail
+
+        def on_complete(env)
+          return unless env.response_headers['content-type'] =~ /\b(xml)/
```
Comment (saneshark) [inline lib/preneeds/middleware/response/eoas_xml_errors.rb]: Won't it always be XML? I think we would want to return from this middleware based on status code instead maybe.

**Comment by saneshark** [comment]
Actually looking at this code this LGTM.

File: lib/preneeds/middleware/response/clean_response.rb (hunk 1 above)
Comment (mphprogrammer) [inline lib/preneeds/middleware/response/clean_response.rb]: Yup - missed that.

File: lib/preneeds/middleware/response/eoas_xml_errors.rb (hunk 2 above)
Comment (mphprogrammer) [inline lib/preneeds/middleware/response/eoas_xml_errors.rb]: Problem is 200 can also be an error.

**Pull request merged**